013-large-sum/src/main.rs

```diff
@@ -0,0 +1,164 @@
+// Project Euler # 13
+// Work out the first ten digits of the sum of the following one-hundred 50-digit numbers.
+
+#[derive(Clone)]
+struct Total {
+    num: Vec<char>,
+}
+
+impl std::fmt::Display for Total {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}", String::from_iter(self.num.clone().iter().rev()))
+    }
+}
+
+impl Total {
+    fn new() -> Total {
+        Total {
+            num: "0".chars().collect(),
+        }
+    }
+
+    fn add(&mut self, add_num: &str) {
+        let max_len = self.num.len().max(add_num.len());
+        let v1 = self
+            .num
+            .clone()
+            .into_iter()
```

Comment:
Does this vector have to be consumed?  You could avoid a clone each request ?  If anything, i would call 

```rust
self.num.iter().copied().chain()...
```
A  `char` implements Copy Which is cheaper then a clone (a pointer copy instead of a deep copy).  This will simply to a bit-wise copy on every iteration you need.

**Reference for Char Copy**:
https://doc.rust-lang.org/std/primitive.char.html#impl-Copy-for-char

You could also use `.cloned()` over `clone()`
```rust
self.num.iter().cloned().chain()...
```

Cloned creates a borrowed iterator over the Vec that yields `&T`.  This does not create a new vector, but borrows each element from the vector, and clones those on the fly, based off `&T`  for `Iterator::next()`.  This means you'll only clone N elements from take instead of hte whole vector.

You should prefer to use `cloned` vs `clone` when iterating over a vector when possible.  There are use cases, such as `into_iter()` that has some cool methods, such as getting the underlying slice of the vector and such.

You should use `copied()` over `cloned()` when possible.